specs/commandline/plugin.md

```diff
@@ -36,15 +36,55 @@ Aliases:
   list, ls
 ```
 
+### notation plugin install
+
+```text
+Installs a plugin
```

Comment:
I like this command but have some questions

1. How will install/uninstall work aka whats your vision for this commands? Will it allow execution of custom steps which might be required for setup/cleanup?
2. what should be the format of plugin package?
3. IMO user doesn't needs to specify plugin name, we can extract it using [plugin metada](https://github.com/notaryproject/notaryproject/blob/main/specs/plugin-extensibility.md#plugin-metadata) command

Reply:
Hey @priteshbandi happy to share my thoughts.

1. The Install subcommand would create the necessary directory under the default plugin path and extract/move the plugin to the directory. The delete subcommand would remove the plugin from the directory. If there's more than needs to happen, please lmk.
2. By format of the plugin package do you mean is compressed or not? I was planning to support both formats and optionally expand the compressed version if needed and move the file into the plugin directory.
3. That's a great idea, is that metadata extractable from the executable?

Reply:
1. IMO just providing copy and delete plugin file would be a good start; I was thinking about following enhancemenets and they can be be future work.
    1. add pre-install/post-install(similar for uninstall) script support since plugin might want to perform setup/cleanup.
    2. On install, verifying signature of plugin and displaying its data to user. This will help user verify authenticity and integrity of plugin.
2. yes, by format I meant compressed or not.
3. Yes we/notation should be able to call that command on the executable.

Reply:
@priteshbandi since the `get-plugin-metadata` subcommand can get the plugin-name I agree that the user shouldn't have to specify the plugin-name. @yizha1 and @shizhMSFT what do you think?

Reply:
@Duffney @priteshbandi I like the idea of getting the plugin name from metadata, so that users don't need to specify it. We need to consider the output of installation command, so that users know the plugin name and other information as well.

Some questions:
- How do we know if two plugin files with different names are the same plugin? What is the desired behavior if users install the same plugin multiple times?
- Users install the same plugin but with a higher version, what is the desired behavior?

Reply:
Here are my thoughts:

- Assuming the plugin naming standard of `notation-plugin-name` is followed you can use a string parser to get the plugin name from the binary and then use the `get-metadata` function to return the plugin details and compare the values. Imo if the user installs the same plugin multiple times the desired behavior should be to have the new plugin overwrite\replace the existing plugin.
- If the user installs the same plugin, the plugin version should be compared then displayed to the user with a prompt confirming they wish to overwrite the existing plugin.